docs/kns_apps_tutorial.md

```diff
@@ -0,0 +1,347 @@
+# Jarvice KNS Applications tutorial
+
+![GlobalProcess](img/apps_tutorial/SuperNimbix.svg)
+
+Note: it is assumed that the user is already familliar with Jarvice Push to Compute (P2C). Please refer to [apps tutorial](apps_tutorial.md).
+
+Note 2: This tutorial assumes user is building applications based on `appdefversion` version `2`.
+
+## 1. Introduction
+
+Before creating apps for KNS scheduler, it is important to understand how KNS operates, and what exactly is the KNS.
+
+KNS stands for Kubernetes Nested Scheduler. As the name says, KNS is a Jarvice downstream component that will spawn nested Kubernetes clusters (based on K3S)
+and will deploy Kubernetes based apps inside these nested clusters, as jobs.
+
+While traditional Jarvice downstreams are made for HPC jobs and parallel short term jobs, KNS has been designed to host long running jobs with massive scaleup capabilities. KNS is in theory capable or running anything that can run on a Kubernetes cluster, making it a very nice shared "sandbox".
+
+KNS apps are docker images that contains files (mostly helm/kubectl/kustomize templates) and instructions to be used to deploy Kubernetes based apps, like Kubeflow, ArgoCD, Kubeai, etc. When launching a new job, the KNS will create a dedicated nested K3S cluster for the job, pull the app image from within this cluster, and use image's instructions to deploy the expected resources. Ingresses will allow user to access the nested cluster services.
+
+Note: some apps example for KNS can be found in the [Nimbix kns-apps repository](https://github.com/nimbix/kns-apps).
```

Comment:
Repository is not public. We should check if we want to open it

Reply:
Good point, I asked if its possible to open it.